lib/Utils/Layout/Codegen.cpp

```diff
@@ -123,5 +146,155 @@ FailureOr<std::string> generateLoopNestAsCStr(
   return actual.insert(0, "\n");
 }
 
+Value buildIslExpr(isl_ast_expr* expr, std::map<std::string, Value> ivToValue,
+                   OpBuilder& builder) {
+  ImplicitLocOpBuilder b(builder.getUnknownLoc(), builder);
```

Comment:
These loops do come from real ops, so we should be able to thread the Location through. I do think this is valuable for tracing and debugging. Maybe pass it through the constructor for the LoopNestGenerator class?

Also, construct the ImplicitLocOpBuilder outside of this method and pass it in. Since you recurse below, it will nest these constructirs repeatedly.